README.md
<h1 align="center">mime-db</h1>

<div align="center">
  <p>
    <strong>Media Type Database, looks up `extension` or `media type`</strong>
  </p>
</div>

<div align="center">
  <!-- Docs.rs docs -->
  <a href="https://docs.rs/mime-db">
    <img src="https://img.shields.io/badge/docs-latest-blue.svg?style=flat-square"
      alt="Docs.rs docs" /></a>
  <!-- Crates version -->
  <a href="https://crates.io/crates/mime-db">
    <img src="https://img.shields.io/crates/v/mime-db.svg?style=flat-square"
    alt="Crates.io version" /></a>
  <!-- Downloads -->
  <a href="https://crates.io/crates/mime-db">
    <img src="https://img.shields.io/crates/d/mime-db.svg?style=flat-square"
      alt="Download" /></a>
  <a href="https://github.com/jshttp/mime-db">
    <img src="https://img.shields.io/npm/v/mime-db/latest?style=flat-square&label=jshttp%2Fmime-db"
      alt="jshttp/mime-db" /></a>
</div>

## Installation

```shell
cargo add mime-db
```

## Usage

```rust
use mime_db::{extension, extensions, lookup};

assert_eq!(lookup("json").unwrap(), "application/json");
assert_eq!(lookup(".md").unwrap(), "text/markdown");
assert_eq!(lookup("folder/file.js").unwrap(), "application/javascript");
assert_eq!(lookup("folder/.htaccess"), None);
assert_eq!(lookup("cats"), None);

assert!(extensions2("application/octet-stream").eq([
    "bin", "dms", "lrf", "mar", "so", "dist", "distz", "pkg", "bpk", "dump", "elc",
    "deploy", "exe", "dll", "deb", "dmg", "iso", "img", "msi", "msp", "msm", "buffer"
]
.iter()
.cloned()));

assert_eq!(extension("application/octet-stream").unwrap(), "bin");
```

## License

<sup>
Licensed under either of <a href="LICENSE-APACHE">Apache License, Version
2.0</a> or <a href="LICENSE-MIT">MIT license</a> at your option.
</sup>

<br>

<sub>
Unless you explicitly state otherwise, any contribution intentionally submitted
for inclusion in this crate by you, as defined in the Apache-2.0 license, shall
be dual licensed as above, without any additional terms or conditions.
</sub>


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [1.7.0] - 2023-06-26

### Changed

- The `extensions2` and `lookup` functions now early exit when an empty input is passed.

- Synced with [`jshttp/mime-db` v1.52.0](https://github.com/jshttp/mime-db/releases/tag/v1.52.0)

## [1.5.0] - 2021-09-11

### Changed

- Synced with [`jshttp/mime-db` v1.49.0](https://github.com/jshttp/mime-db/releases/tag/v1.49.0)

## [1.4.0] - 2021-06-03

### Changed

- Synced with [`jshttp/mime-db` v1.48.0](https://github.com/jshttp/mime-db/releases/tag/v1.48.0)

## [1.3.0] - 2021-04-06

### Changed

- Synced with [`jshttp/mime-db` v1.47.0](https://github.com/jshttp/mime-db/releases/tag/v1.47.0)

## [1.2.0] - 2021-02-19

### Changed

- Synced with [`jshttp/mime-db` v1.46.0](https://github.com/jshttp/mime-db/releases/tag/v1.46.0)

## [1.1.0] - 2020-10-06

### Added

- `extensions2`

### Changed

- Synced with [`jshttp/mime-db` v1.45.0](https://github.com/jshttp/mime-db/releases/tag/v1.45.0)

[Unreleased]: https://github.com/viz-rs/mime-db/compare/v1.7.0...HEAD
[1.7.0]: https://github.com/viz-rs/mime-db/compare/v1.6.0...v1.k.0
[1.6.0]: https://github.com/viz-rs/mime-db/compare/v1.5.0...v1.6.0
[1.5.0]: https://github.com/viz-rs/mime-db/compare/v1.4.0...v1.5.0

src/extensions.rs
pub const EXTENSIONS: [(&str, usize); 1220] = [
    ("ez", 0),
    ("aw", 1),
    ("atom", 2),
    ("atomcat", 3),
    ("atomdeleted", 4),
    ("atomsvc", 5),
    ("dwd", 6),
    ("held", 7),
    ("rsat", 8),
    ("bdoc", 9),
    ("xcs", 10),
    ("ccxml", 11),
    ("cdfx", 12),
    ("cdmia", 13),
    ("cdmic", 14),
    ("cdmid", 15),
    ("cdmio", 16),
    ("cdmiq", 17),
    ("cpl", 18),
    ("cu", 19),
    ("mpd", 20),
    ("mpp", 21),
    ("davmount", 22),
    ("dbk", 23),
    ("dssc", 24),
    ("xdssc", 25),
    ("es", 26),
    ("ecma", 26),
    ("emma", 27),
    ("emotionml", 28),
    ("epub", 29),
    ("exi", 30),
    ("exp", 31),
    ("fdt", 32),
    ("pfr", 33),
    ("geojson", 34),
    ("gml", 35),
    ("gpx", 36),
    ("gxf", 37),
    ("gz", 38),
    ("hjson", 39),
    ("stk", 40),
    ("ink", 41),
    ("inkml", 41),
    ("ipfix", 42),
    ("its", 43),
    ("jar", 44),
    ("war", 44),
    ("ear", 44),
    ("ser", 45),
    ("class", 46),
    ("js", 47),
    ("mjs", 47),
    ("json", 48),
    ("map", 48),
    ("json5", 49),
    ("jsonml", 50),
    ("jsonld", 51),
    ("lgr", 52),
    ("lostxml", 53),
    ("hqx", 54),
    ("cpt", 55),
    ("mads", 56),
    ("webmanifest", 57),
    ("mrc", 58),
    ("mrcx", 59),
    ("ma", 60),
    ("nb", 60),
    ("mb", 60),
    ("mathml", 61),
    ("mbox", 62),
    ("mpf", 63),
    ("mscml", 64),
    ("metalink", 65),
    ("meta4", 66),
    ("mets", 67),
    ("maei", 68),
    ("musd", 69),
    ("mods", 70),
    ("m21", 71),
    ("mp21", 71),
    ("mp4s", 72),
    ("m4p", 72),
    ("doc", 73),
    ("dot", 73),
    ("mxf", 74),
    ("nq", 75),
    ("nt", 76),
    ("cjs", 77),
    ("bin", 78),
    ("dms", 78),
    ("lrf", 78),
    ("mar", 78),
    ("so", 78),
    ("dist", 78),
    ("distz", 78),
    ("pkg", 78),
    ("bpk", 78),
    ("dump", 78),
    ("elc", 78),
    ("deploy", 78),
    ("exe", 78),
    ("dll", 78),
    ("deb", 78),
    ("dmg", 78),
    ("iso", 78),
    ("img", 78),
    ("msi", 78),
    ("msp", 78),
    ("msm", 78),
    ("buffer", 78),
    ("oda", 79),
    ("opf", 80),
    ("ogx", 81),
    ("omdoc", 82),
    ("onetoc", 83),
    ("onetoc2", 83),
    ("onetmp", 83),
    ("onepkg", 83),
    ("oxps", 84),
    ("relo", 85),
    ("xer", 86),
    ("pdf", 87),
    ("pgp", 88),
    ("asc", 89),
    ("asc", 90),
    ("sig", 90),
    ("prf", 91),
    ("p10", 92),
    ("p7m", 93),
    ("p7c", 93),
    ("p7s", 94),
    ("p8", 95),
    ("ac", 96),
    ("cer", 97),
    ("crl", 98),
    ("pkipath", 99),
    ("pki", 100),
    ("pls", 101),
    ("ai", 102),
    ("eps", 102),
    ("ps", 102),
    ("provx", 103),
    ("cww", 104),
    ("pskcxml", 105),
    ("raml", 106),
    ("rdf", 107),
    ("owl", 107),
    ("rif", 108),
    ("rnc", 109),
    ("rl", 110),
    ("rld", 111),
    ("rs", 112),
    ("rapd", 113),
    ("sls", 114),
    ("rusd", 115),
    ("gbr", 116),
    ("mft", 117),
    ("roa", 118),
    ("rsd", 119),
    ("rss", 120),
    ("rtf", 121),
    ("sbml", 122),
    ("scq", 123),
    ("scs", 124),
    ("spq", 125),
    ("spp", 126),
    ("sdp", 127),
    ("senmlx", 128),
    ("sensmlx", 129),
    ("setpay", 130),
    ("setreg", 131),
    ("shf", 132),
    ("siv", 133),
    ("sieve", 133),
    ("smi", 134),
    ("smil", 134),
    ("rq", 135),
    ("srx", 136),
    ("gram", 137),
    ("grxml", 138),
    ("sru", 139),
    ("ssdl", 140),
    ("ssml", 141),
    ("swidtag", 142),
    ("tei", 143),
    ("teicorpus", 143),
    ("tfi", 144),
    ("tsd", 145),
    ("toml", 146),
    ("trig", 147),
    ("ttml", 148),
    ("ubj", 149),
    ("rsheet", 150),
    ("td", 151),
    ("1km", 152),
    ("plb", 153),
    ("psb", 154),
    ("pvb", 155),
    ("tcap", 156),
    ("pwn", 157),
    ("aso", 158),
    ("imp", 159),
    ("acu", 160),
    ("atc", 161),
    ("acutc", 161),
    ("air", 162),
    ("fcdt", 163),
    ("fxp", 164),
    ("fxpl", 164),
    ("xdp", 165),
    ("xfdf", 166),
    ("age", 167),
    ("ahead", 168),
    ("azf", 169),
    ("azs", 170),
    ("azw", 171),
    ("acc", 172),
    ("ami", 173),
    ("apk", 174),
    ("cii", 175),
    ("fti", 176),
    ("atx", 177),
    ("mpkg", 178),
    ("key", 179),
    ("m3u8", 180),
    ("numbers", 181),
    ("pages", 182),
    ("pkpass", 183),
    ("swi", 184),
    ("iota", 185),
    ("aep", 186),
    ("bmml", 187),
    ("mpm", 188),
    ("bmi", 189),
    ("rep", 190),
    ("cdxml", 191),
    ("mmd", 192),
    ("cdy", 193),
    ("csl", 194),
    ("cla", 195),
    ("rp9", 196),
    ("c4g", 197),
    ("c4d", 197),
    ("c4f", 197),
    ("c4p", 197),
    ("c4u", 197),
    ("c11amc", 198),
    ("c11amz", 199),
    ("csp", 200),
    ("cdbcmsg", 201),
    ("cmc", 202),
    ("clkx", 203),
    ("clkk", 204),
    ("clkp", 205),
    ("clkt", 206),
    ("clkw", 207),
    ("wbs", 208),
    ("pml", 209),
    ("ppd", 210),
    ("car", 211),
    ("pcurl", 212),
    ("dart", 213),
    ("rdz", 214),
    ("dbf", 215),
    ("uvf", 216),
    ("uvvf", 216),
    ("uvd", 216),
    ("uvvd", 216),
    ("uvt", 217),
    ("uvvt", 217),
    ("uvx", 218),
    ("uvvx", 218),
    ("uvz", 219),
    ("uvvz", 219),
    ("fe_launch", 220),
    ("dna", 221),
    ("mlp", 222),
    ("dpg", 223),
    ("dfac", 224),
    ("kpxx", 225),
    ("ait", 226),
    ("svc", 227),
    ("geo", 228),
    ("mag", 229),
    ("nml", 230),
    ("esf", 231),
    ("msf", 232),
    ("qam", 233),
    ("slt", 234),
    ("ssf", 235),
    ("es3", 236),
    ("et3", 236),
    ("ez2", 237),
    ("ez3", 238),
    ("fdf", 239),
    ("mseed", 240),
    ("seed", 241),
    ("dataless", 241),
    ("gph", 242),
    ("ftc", 243),
    ("fm", 244),
    ("frame", 244),
    ("maker", 244),
    ("book", 244),
    ("fnc", 245),
    ("ltf", 246),
    ("fsc", 247),
    ("oas", 248),
    ("oa2", 249),
    ("oa3", 250),
    ("fg5", 251),
    ("bh2", 252),
    ("ddd", 253),
    ("xdw", 254),
    ("xbd", 255),
    ("fzs", 256),
    ("txd", 257),
    ("ggb", 258),
    ("ggt", 259),
    ("gex", 260),
    ("gre", 260),
    ("gxt", 261),
    ("g2w", 262),
    ("g3w", 263),
    ("gmx", 264),
    ("gdoc", 265),
    ("gslides", 266),
    ("gsheet", 267),
    ("kml", 268),
    ("kmz", 269),
    ("gqf", 270),
    ("gqs", 270),
    ("gac", 271),
    ("ghf", 272),
    ("gim", 273),
    ("grv", 274),
    ("gtm", 275),
    ("tpl", 276),
    ("vcg", 277),
    ("hal", 278),
    ("zmm", 279),
    ("hbci", 280),
    ("les", 281),
    ("hpgl", 282),
    ("hpid", 283),
    ("hps", 284),
    ("jlt", 285),
    ("pcl", 286),
    ("pclxl", 287),
    ("sfd-hdstx", 288),
    ("mpy", 289),
    ("afp", 290),
    ("listafp", 290),
    ("list3820", 290),
    ("irm", 291),
    ("sc", 292),
    ("icc", 293),
    ("icm", 293),
    ("igl", 294),
    ("ivp", 295),
    ("ivu", 296),
    ("igm", 297),
    ("xpw", 298),
    ("xpx", 298),
    ("i2g", 299),
    ("qbo", 300),
    ("qfx", 301),
    ("rcprofile", 302),
    ("irp", 303),
    ("xpr", 304),
    ("fcs", 305),
    ("jam", 306),
    ("rms", 307),
    ("jisp", 308),
    ("joda", 309),
    ("ktz", 310),
    ("ktr", 310),
    ("karbon", 311),
    ("chrt", 312),
    ("kfo", 313),
    ("flw", 314),
    ("kon", 315),
    ("kpr", 316),
    ("kpt", 316),
    ("ksp", 317),
    ("kwd", 318),
    ("kwt", 318),
    ("htke", 319),
    ("kia", 320),
    ("kne", 321),
    ("knp", 321),
    ("skp", 322),
    ("skd", 322),
    ("skt", 322),
    ("skm", 322),
    ("sse", 323),
    ("lasxml", 324),
    ("lbd", 325),
    ("lbe", 326),
    ("123", 327),
    ("apr", 328),
    ("pre", 329),
    ("nsf", 330),
    ("org", 331),
    ("scm", 332),
    ("lwp", 333),
    ("portpkg", 334),
    ("mvt", 335),
    ("mcd", 336),
    ("mc1", 337),
    ("cdkey", 338),
    ("mwf", 339),
    ("mfm", 340),
    ("flo", 341),
    ("igx", 342),
    ("mif", 343),
    ("daf", 344),
    ("dis", 345),
    ("mbk", 346),
    ("mqy", 347),
    ("msl", 348),
    ("plc", 349),
    ("txf", 350),
    ("mpn", 351),
    ("mpc", 352),
    ("xul", 353),
    ("cil", 354),
    ("cab", 355),
    ("xls", 356),
    ("xlm", 356),
    ("xla", 356),
    ("xlc", 356),
    ("xlt", 356),
    ("xlw", 356),
    ("xlam", 357),
    ("xlsb", 358),
    ("xlsm", 359),
    ("xltm", 360),
    ("eot", 361),
    ("chm", 362),
    ("ims", 363),
    ("lrm", 364),
    ("thmx", 365),
    ("msg", 366),
    ("cat", 367),
    ("stl", 368),
    ("ppt", 369),
    ("pps", 369),
    ("pot", 369),
    ("ppam", 370),
    ("pptm", 371),
    ("sldm", 372),
    ("ppsm", 373),
    ("potm", 374),
    ("mpp", 375),
    ("mpt", 375),
    ("docm", 376),
    ("dotm", 377),
    ("wps", 378),
    ("wks", 378),
    ("wcm", 378),
    ("wdb", 378),
    ("wpl", 379),
    ("xps", 380),
    ("mseq", 381),
    ("mus", 382),
    ("msty", 383),
    ("taglet", 384),
    ("nlu", 385),
    ("ntf", 386),
    ("nitf", 386),
    ("nnd", 387),
    ("nns", 388),
    ("nnw", 389),
    ("ac", 390),
    ("ngdat", 391),
    ("n-gage", 392),
    ("rpst", 393),
    ("rpss", 394),
    ("edm", 395),
    ("edx", 396),
    ("ext", 397),
    ("odc", 398),
    ("otc", 399),
    ("odb", 400),
    ("odf", 401),
    ("odft", 402),
    ("odg", 403),
    ("otg", 404),
    ("odi", 405),
    ("oti", 406),
    ("odp", 407),
    ("otp", 408),
    ("ods", 409),
    ("ots", 410),
    ("odt", 411),
    ("odm", 412),
    ("ott", 413),
    ("oth", 414),
    ("xo", 415),
    ("dd2", 416),
    ("obgx", 417),
    ("oxt", 418),
    ("osm", 419),
    ("pptx", 420),
    ("sldx", 421),
    ("ppsx", 422),
    ("potx", 423),
    ("xlsx", 424),
    ("xltx", 425),
    ("docx", 426),
    ("dotx", 427),
    ("mgp", 428),
    ("dp", 429),
    ("esa", 430),
    ("pdb", 431),
    ("pqa", 431),
    ("oprc", 431),
    ("paw", 432),
    ("str", 433),
    ("ei6", 434),
    ("efif", 435),
    ("wg", 436),
    ("plf", 437),
    ("pbd", 438),
    ("box", 439),
    ("mgz", 440),
    ("qps", 441),
    ("ptid", 442),
    ("qxd", 443),
    ("qxt", 443),
    ("qwd", 443),
    ("qwt", 443),
    ("qxl", 443),
    ("qxb", 443),
    ("rar", 444),
    ("bed", 445),
    ("mxl", 446),
    ("musicxml", 447),
    ("cryptonote", 448),
    ("cod", 449),
    ("rm", 450),
    ("rmvb", 451),
    ("link66", 452),
    ("st", 453),
    ("see", 454),
    ("sema", 455),
    ("semd", 456),
    ("semf", 457),
    ("ifm", 458),
    ("itp", 459),
    ("iif", 460),
    ("ipk", 461),
    ("twd", 462),
    ("twds", 462),
    ("mmf", 463),
    ("teacher", 464),
    ("fo", 465),
    ("sdkm", 466),
    ("sdkd", 466),
    ("dxp", 467),
    ("sfs", 468),
    ("sdc", 469),
    ("sda", 470),
    ("sdd", 471),
    ("smf", 472),
    ("sdw", 473),
    ("vor", 473),
    ("sgl", 474),
    ("smzip", 475),
    ("sm", 476),
    ("wadl", 477),
    ("sxc", 478),
    ("stc", 479),
    ("sxd", 480),
    ("std", 481),
    ("sxi", 482),
    ("sti", 483),
    ("sxm", 484),
    ("sxw", 485),
    ("sxg", 486),
    ("stw", 487),
    ("sus", 488),
    ("susp", 488),
    ("svd", 489),
    ("sis", 490),
    ("sisx", 490),
    ("xsm", 491),
    ("bdm", 492),
    ("xdm", 493),
    ("ddf", 494),
    ("tao", 495),
    ("pcap", 496),
    ("cap", 496),
    ("dmp", 496),
    ("tmo", 497),
    ("tpt", 498),
    ("mxs", 499),
    ("tra", 500),
    ("ufd", 501),
    ("ufdl", 501),
    ("utz", 502),
    ("umj", 503),
    ("unityweb", 504),
    ("uoml", 505),
    ("vcx", 506),
    ("vsd", 507),
    ("vst", 507),
    ("vss", 507),
    ("vsw", 507),
    ("vis", 508),
    ("vsf", 509),
    ("wbxml", 510),
    ("wmlc", 511),
    ("wmlsc", 512),
    ("wtb", 513),
    ("nbp", 514),
    ("wpd", 515),
    ("wqd", 516),
    ("stf", 517),
    ("xar", 518),
    ("xfdl", 519),
    ("hvd", 520),
    ("hvs", 521),
    ("hvp", 522),
    ("osf", 523),
    ("osfpvg", 524),
    ("saf", 525),
    ("spf", 526),
    ("cmp", 527),
    ("zir", 528),
    ("zirz", 528),
    ("zaz", 529),
    ("vxml", 530),
    ("wasm", 531),
    ("wif", 532),
    ("wgt", 533),
    ("hlp", 534),
    ("wsdl", 535),
    ("wspolicy", 536),
    ("7z", 537),
    ("abw", 538),
    ("ace", 539),
    ("dmg", 540),
    ("arj", 541),
    ("aab", 542),
    ("x32", 542),
    ("u32", 542),
    ("vox", 542),
    ("aam", 543),
    ("aas", 544),
    ("bcpio", 545),
    ("bdoc", 546),
    ("torrent", 547),
    ("blb", 548),
    ("blorb", 548),
    ("bz", 549),
    ("bz2", 550),
    ("boz", 550),
    ("cbr", 551),
    ("cba", 551),
    ("cbt", 551),
    ("cbz", 551),
    ("cb7", 551),
    ("vcd", 552),
    ("cfs", 553),
    ("chat", 554),
    ("pgn", 555),
    ("crx", 556),
    ("cco", 557),
    ("nsc", 558),
    ("cpio", 559),
    ("csh", 560),
    ("deb", 561),
    ("udeb", 561),
    ("dgc", 562),
    ("dir", 563),
    ("dcr", 563),
    ("dxr", 563),
    ("cst", 563),
    ("cct", 563),
    ("cxt", 563),
    ("w3d", 563),
    ("fgd", 563),
    ("swa", 563),
    ("wad", 564),
    ("ncx", 565),
    ("dtb", 566),
    ("res", 567),
    ("dvi", 568),
    ("evy", 569),
    ("eva", 570),
    ("bdf", 571),
    ("gsf", 572),
    ("psf", 573),
    ("pcf", 574),
    ("snf", 575),
    ("pfa", 576),
    ("pfb", 576),
    ("pfm", 576),
    ("afm", 576),
    ("arc", 577),
    ("spl", 578),
    ("gca", 579),
    ("ulx", 580),
    ("gnumeric", 581),
    ("gramps", 582),
    ("gtar", 583),
    ("hdf", 584),
    ("php", 585),
    ("install", 586),
    ("iso", 587),
    ("key", 588),
    ("numbers", 589),
    ("pages", 590),
    ("jardiff", 591),
    ("jnlp", 592),
    ("kdbx", 593),
    ("latex", 594),
    ("luac", 595),
    ("lzh", 596),
    ("lha", 596),
    ("run", 597),
    ("mie", 598),
    ("prc", 599),
    ("mobi", 599),
    ("application", 600),
    ("lnk", 601),
    ("wmd", 602),
    ("wmz", 603),
    ("xbap", 604),
    ("mdb", 605),
    ("obd", 606),
    ("crd", 607),
    ("clp", 608),
    ("exe", 609),
    ("exe", 610),
    ("dll", 610),
    ("com", 610),
    ("bat", 610),
    ("msi", 610),
    ("mvb", 611),
    ("m13", 611),
    ("m14", 611),
    ("wmf", 612),
    ("wmz", 612),
    ("emf", 612),
    ("emz", 612),
    ("mny", 613),
    ("pub", 614),
    ("scd", 615),
    ("trm", 616),
    ("wri", 617),
    ("nc", 618),
    ("cdf", 618),
    ("pac", 619),
    ("nzb", 620),
    ("pl", 621),
    ("pm", 621),
    ("prc", 622),
    ("pdb", 622),
    ("p12", 623),
    ("pfx", 623),
    ("p7b", 624),
    ("spc", 624),
    ("p7r", 625),
    ("rar", 626),
    ("rpm", 627),
    ("ris", 628),
    ("sea", 629),
    ("sh", 630),
    ("shar", 631),
    ("swf", 632),
    ("xap", 633),
    ("sql", 634),
    ("sit", 635),
    ("sitx", 636),
    ("srt", 637),
    ("sv4cpio", 638),
    ("sv4crc", 639),
    ("t3", 640),
    ("gam", 641),
    ("tar", 642),
    ("tcl", 643),
    ("tk", 643),
    ("tex", 644),
    ("tfm", 645),
    ("texinfo", 646),
    ("texi", 646),
    ("obj", 647),
    ("ustar", 648),
    ("hdd", 649),
    ("ova", 650),
    ("ovf", 651),
    ("vbox", 652),
    ("vbox-extpack", 653),
    ("vdi", 654),
    ("vhd", 655),
    ("vmdk", 656),
    ("src", 657),
    ("webapp", 658),
    ("der", 659),
    ("crt", 659),
    ("pem", 659),
    ("fig", 660),
    ("xlf", 661),
    ("xpi", 662),
    ("xz", 663),
    ("z1", 664),
    ("z2", 664),
    ("z3", 664),
    ("z4", 664),
    ("z5", 664),
    ("z6", 664),
    ("z7", 664),
    ("z8", 664),
    ("xaml", 665),
    ("xav", 666),
    ("xca", 667),
    ("xdf", 668),
    ("xel", 669),
    ("xns", 670),
    ("xenc", 671),
    ("xhtml", 672),
    ("xht", 672),
    ("xlf", 673),
    ("xml", 674),
    ("xsl", 674),
    ("xsd", 674),
    ("rng", 674),
    ("dtd", 675),
    ("xop", 676),
    ("xpl", 677),
    ("xsl", 678),
    ("xslt", 678),
    ("xspf", 679),
    ("mxml", 680),
    ("xhvml", 680),
    ("xvml", 680),
    ("xvm", 680),
    ("yang", 681),
    ("yin", 682),
    ("zip", 683),
    ("3gpp", 684),
    ("adp", 685),
    ("amr", 686),
    ("au", 687),
    ("snd", 687),
    ("mid", 688),
    ("midi", 688),
    ("kar", 688),
    ("rmi", 688),
    ("mxmf", 689),
    ("mp3", 690),
    ("m4a", 691),
    ("mp4a", 691),
    ("mpga", 692),
    ("mp2", 692),
    ("mp2a", 692),
    ("mp3", 692),
    ("m2a", 692),
    ("m3a", 692),
    ("oga", 693),
    ("ogg", 693),
    ("spx", 693),
    ("opus", 693),
    ("s3m", 694),
    ("sil", 695),
    ("uva", 696),
    ("uvva", 696),
    ("eol", 697),
    ("dra", 698),
    ("dts", 699),
    ("dtshd", 700),
    ("lvp", 701),
    ("pya", 702),
    ("ecelp4800", 703),
    ("ecelp7470", 704),
    ("ecelp9600", 705),
    ("rip", 706),
    ("wav", 707),
    ("wav", 708),
    ("weba", 709),
    ("aac", 710),
    ("aif", 711),
    ("aiff", 711),
    ("aifc", 711),
    ("caf", 712),
    ("flac", 713),
    ("m4a", 714),
    ("mka", 715),
    ("m3u", 716),
    ("wax", 717),
    ("wma", 718),
    ("ram", 719),
    ("ra", 719),
    ("rmp", 720),
    ("ra", 721),
    ("wav", 722),
    ("xm", 723),
    ("cdx", 724),
    ("cif", 725),
    ("cmdf", 726),
    ("cml", 727),
    ("csml", 728),
    ("xyz", 729),
    ("ttc", 730),
    ("otf", 731),
    ("ttf", 732),
    ("woff", 733),
    ("woff2", 734),
    ("exr", 735),
    ("apng", 736),
    ("avci", 737),
    ("avcs", 738),
    ("avif", 739),
    ("bmp", 740),
    ("cgm", 741),
    ("drle", 742),
    ("emf", 743),
    ("fits", 744),
    ("g3", 745),
    ("gif", 746),
    ("heic", 747),
    ("heics", 748),
    ("heif", 749),
    ("heifs", 750),
    ("hej2", 751),
    ("hsj2", 752),
    ("ief", 753),
    ("jls", 754),
    ("jp2", 755),
    ("jpg2", 755),
    ("jpeg", 756),
    ("jpg", 756),
    ("jpe", 756),
    ("jph", 757),
    ("jhc", 758),
    ("jpm", 759),
    ("jpx", 760),
    ("jpf", 760),
    ("jxr", 761),
    ("jxra", 762),
    ("jxrs", 763),
    ("jxs", 764),
    ("jxsc", 765),
    ("jxsi", 766),
    ("jxss", 767),
    ("ktx", 768),
    ("ktx2", 769),
    ("png", 770),
    ("btif", 771),
    ("pti", 772),
    ("sgi", 773),
    ("svg", 774),
    ("svgz", 774),
    ("t38", 775),
    ("tif", 776),
    ("tiff", 776),
    ("tfx", 777),
    ("psd", 778),
    ("azv", 779),
    ("uvi", 780),
    ("uvvi", 780),
    ("uvg", 780),
    ("uvvg", 780),
    ("djvu", 781),
    ("djv", 781),
    ("sub", 782),
    ("dwg", 783),
    ("dxf", 784),
    ("fbs", 785),
    ("fpx", 786),
    ("fst", 787),
    ("mmr", 788),
    ("rlc", 789),
    ("ico", 790),
    ("dds", 791),
    ("mdi", 792),
    ("wdp", 793),
    ("npx", 794),
    ("b16", 795),
    ("tap", 796),
    ("vtf", 797),
    ("wbmp", 798),
    ("xif", 799),
    ("pcx", 800),
    ("webp", 801),
    ("wmf", 802),
    ("3ds", 803),
    ("ras", 804),
    ("cmx", 805),
    ("fh", 806),
    ("fhc", 806),
    ("fh4", 806),
    ("fh5", 806),
    ("fh7", 806),
    ("ico", 807),
    ("jng", 808),
    ("sid", 809),
    ("bmp", 810),
    ("pcx", 811),
    ("pic", 812),
    ("pct", 812),
    ("pnm", 813),
    ("pbm", 814),
    ("pgm", 815),
    ("ppm", 816),
    ("rgb", 817),
    ("tga", 818),
    ("xbm", 819),
    ("xpm", 820),
    ("xwd", 821),
    ("disposition-notification", 822),
    ("u8msg", 823),
    ("u8dsn", 824),
    ("u8mdn", 825),
    ("u8hdr", 826),
    ("eml", 827),
    ("mime", 827),
    ("wsc", 828),
    ("3mf", 829),
    ("gltf", 830),
    ("glb", 831),
    ("igs", 832),
    ("iges", 832),
    ("msh", 833),
    ("mesh", 833),
    ("silo", 833),
    ("mtl", 834),
    ("obj", 835),
    ("stpx", 836),
    ("stpz", 837),
    ("stpxz", 838),
    ("stl", 839),
    ("dae", 840),
    ("dwf", 841),
    ("gdl", 842),
    ("gtw", 843),
    ("mts", 844),
    ("ogex", 845),
    ("x_b", 846),
    ("x_t", 847),
    ("vds", 848),
    ("usdz", 849),
    ("bsp", 850),
    ("vtu", 851),
    ("wrl", 852),
    ("vrml", 852),
    ("x3db", 853),
    ("x3dbz", 853),
    ("x3db", 854),
    ("x3dv", 855),
    ("x3dvz", 855),
    ("x3d", 856),
    ("x3dz", 856),
    ("x3dv", 857),
    ("appcache", 858),
    ("manifest", 858),
    ("ics", 859),
    ("ifb", 859),
    ("coffee", 860),
    ("litcoffee", 860),
    ("css", 861),
    ("csv", 862),
    ("html", 863),
    ("htm", 863),
    ("shtml", 863),
    ("jade", 864),
    ("jsx", 865),
    ("less", 866),
    ("markdown", 867),
    ("md", 867),
    ("mml", 868),
    ("mdx", 869),
    ("n3", 870),
    ("txt", 871),
    ("text", 871),
    ("conf", 871),
    ("def", 871),
    ("list", 871),
    ("log", 871),
    ("in", 871),
    ("ini", 871),
    ("dsc", 872),
    ("rtx", 873),
    ("rtf", 874),
    ("sgml", 875),
    ("sgm", 875),
    ("shex", 876),
    ("slim", 877),
    ("slm", 877),
    ("spdx", 878),
    ("stylus", 879),
    ("styl", 879),
    ("tsv", 880),
    ("t", 881),
    ("tr", 881),
    ("roff", 881),
    ("man", 881),
    ("me", 881),
    ("ms", 881),
    ("ttl", 882),
    ("uri", 883),
    ("uris", 883),
    ("urls", 883),
    ("vcard", 884),
    ("curl", 885),
    ("dcurl", 886),
    ("mcurl", 887),
    ("scurl", 888),
    ("sub", 889),
    ("ged", 890),
    ("fly", 891),
    ("flx", 892),
    ("gv", 893),
    ("3dml", 894),
    ("spot", 895),
    ("jad", 896),
    ("wml", 897),
    ("wmls", 898),
    ("vtt", 899),
    ("s", 900),
    ("asm", 900),
    ("c", 901),
    ("cc", 901),
    ("cxx", 901),
    ("cpp", 901),
    ("h", 901),
    ("hh", 901),
    ("dic", 901),
    ("htc", 902),
    ("f", 903),
    ("for", 903),
    ("f77", 903),
    ("f90", 903),
    ("hbs", 904),
    ("java", 905),
    ("lua", 906),
    ("mkd", 907),
    ("nfo", 908),
    ("opml", 909),
    ("org", 910),
    ("p", 911),
    ("pas", 911),
    ("pde", 912),
    ("sass", 913),
    ("scss", 914),
    ("etx", 915),
    ("sfv", 916),
    ("ymp", 917),
    ("uu", 918),
    ("vcs", 919),
    ("vcf", 920),
    ("xml", 921),
    ("yaml", 922),
    ("yml", 922),
    ("3gp", 923),
    ("3gpp", 923),
    ("3g2", 924),
    ("h261", 925),
    ("h263", 926),
    ("h264", 927),
    ("m4s", 928),
    ("jpgv", 929),
    ("jpm", 930),
    ("jpgm", 930),
    ("mj2", 931),
    ("mjp2", 931),
    ("ts", 932),
    ("mp4", 933),
    ("mp4v", 933),
    ("mpg4", 933),
    ("mpeg", 934),
    ("mpg", 934),
    ("mpe", 934),
    ("m1v", 934),
    ("m2v", 934),
    ("ogv", 935),
    ("qt", 936),
    ("mov", 936),
    ("uvh", 937),
    ("uvvh", 937),
    ("uvm", 938),
    ("uvvm", 938),
    ("uvp", 939),
    ("uvvp", 939),
    ("uvs", 940),
    ("uvvs", 940),
    ("uvv", 941),
    ("uvvv", 941),
    ("dvb", 942),
    ("fvt", 943),
    ("mxu", 944),
    ("m4u", 944),
    ("pyv", 945),
    ("uvu", 946),
    ("uvvu", 946),
    ("viv", 947),
    ("webm", 948),
    ("f4v", 949),
    ("fli", 950),
    ("flv", 951),
    ("m4v", 952),
    ("mkv", 953),
    ("mk3d", 953),
    ("mks", 953),
    ("mng", 954),
    ("asf", 955),
    ("asx", 955),
    ("vob", 956),
    ("wm", 957),
    ("wmv", 958),
    ("wmx", 959),
    ("wvx", 960),
    ("avi", 961),
    ("movie", 962),
    ("smv", 963),
    ("ice", 964),
];
